File: Subprogramas/Sobrecarga de Operadores/Sobrecarga.cs
//C# (C Sharp_
using System;

class Ponto {
    public int x, y;

    public Ponto(int x, int y) {
        this.x = x;
        this.y = y;
    }

    // Sobrecarga do operador +
    public static Ponto operator +(Ponto p1, Ponto p2) {
        return new Ponto(p1.x + p2.x, p1.y + p2.y);
    }
}

class Sobrecarga {
    static void Main(string[] args) {
        Ponto p1 = new Ponto(2, 3);
        Ponto p2 = new Ponto(4, 5);
        // Usando o operador + sobrecarregado
		Ponto p3 = p1 + p2;  

        Console.WriteLine($"p3 = ({p3.x}, {p3.y})");
    }
}

File: Subprogramas/Sobrecarga de Operadores/sobrecarga.rs
//Rust

use std::ops::Add;

#[derive(Debug)]
struct Ponto {
    x: i32,
    y: i32,
}

// Implementando a sobrecarga do operador +
impl Add for Ponto {
    type Output = Ponto;

    fn add(self, outro: Ponto) -> Ponto {
        Ponto {
            x: self.x + outro.x,
            y: self.y + outro.y,
        }
    }
}

fn main() {
    let p1 = Ponto { x: 2, y: 3 };
    let p2 = Ponto { x: 4, y: 5 };
	// Usando o operador + sobrecarregado
    let p3 = p1 + p2;  

    println!("p3 = ({}, {})", p3.x, p3.y);
}
